music_assistant/controllers/media/audiobooks.py

```diff
@@ -184,7 +184,7 @@ async def _update_library_item(
                 "narrators": serialize_to_json(
                     update.narrators if overwrite else cur_item.narrators or update.narrators
                 ),
-                "duration": update.duration or update.duration,
+                "duration": update.duration if overwrite else cur_item.duration or update.duration,
```

Comment:
Normally you need to put this in a dedicated bugfix PR - then it can also be backported to stable if needed.
For now lets allow it